File: frontend/src/types/quote.ts
export type Sentiment = 'Positive' | 'Negative' | 'Neutral' | 'Frustrated';
export type QuoteType = 'Pain Point' | 'Insight' | 'Feature Request' | 'Praise';

export interface Quote {
  id: string;
  created_time: string;
  quote_text: string;
  speaker: string;
  sentiment: Sentiment;
  quote_type: QuoteType;
  related_persona_id: string | null;
  related_issue_id: string | null;
  competitor_mentioned_id: string | null;
}

File: backend/models/quote.py
from typing import Literal

from pydantic import BaseModel


Sentiment = Literal["Positive", "Negative", "Neutral", "Frustrated"]
QuoteType = Literal["Pain Point", "Insight", "Feature Request", "Praise"]


class QuoteCreate(BaseModel):
    quote_text: str
    speaker: str
    sentiment: Sentiment
    quote_type: QuoteType
    related_persona_id: str | None = None
    related_issue_id: str | None = None
    competitor_mentioned_id: str | None = None


class Quote(QuoteCreate):
    id: str
    created_time: str
